go/libraries/doltcore/sqle/database.go

```diff
@@ -129,6 +129,7 @@ type Database struct {
 	rsw       env.RepoStateWriter
 	batchMode commitBehavior
 	tc        *tableCache
+	tes       *doltdb.TableEditSession
```

Comment:
This should be in the DoltSession, not the database

Reply:
Any real reason why? I looked at `DoltSession`, and it handles things like multiple databases. AFAICT, an edit session should be local to a database, thus it seems to make more sense here?

Reply:
Few reasons:
1) Mixing metaphors: session data belongs in the session object, not the database object

2) The database object isn't supposed to have any state that would prevent it from being used by another session. Like, the DB doesn't have its own root value, it gets it from the session. The one kind-of exception is the table cache, but even that is keyed by root value (which is immutable).

3) The session is in charge of managing things like the current root value for each database for the session. The table editor session has a root value. We will quickly get to a place where the DB session needs to manage this for the editor session.

You are correct that the DoltSession keys everything by database. The EditSession will have to be keyed by database as well.

(If this is going to be a lot of work, you can do it in a follow-up PR, just don't leave it long)